svir/dialogs/load_gmf_data_as_layer_dialog.py

```diff
@@ -64,8 +64,39 @@ def __init__(self, iface, viewer_dock, session, hostname, calc_id,
     def set_ok_button(self):
         self.ok_button.setEnabled(self.imt_cbx.currentIndex() != -1)
 
-    def on_rlz_or_stat_changed(self):
-        self.dataset = self.npz_file[self.rlz_or_stat_cbx.currentText()]
+    def populate_out_dep_widgets(self):
+        self.populate_gmpe_cbx()
+        self.show_num_sites()
+
+    def _format_rlz(self, gmpe_idx):
+        # NOTE: assuming that rlzs are always in the format rlz-XXX
+        return "rlz-%03d" % gmpe_idx
+
+    def show_num_sites(self):
+        # NOTE: we are assuming all realizations have the same number of sites,
+        #       which currently is always true.
+        #       If different realizations have a different number of sites, we
+        #       need to move this block of code inside on_rlz_or_stat_changed()
+        rlz = self._format_rlz(self.gmpe_cbx.currentIndex())
+        gmpe_data = self.npz_file[rlz]
+        self.num_sites_lbl.setText(
+            self.num_sites_msg % gmpe_data.shape)
+
+    def populate_gmpe_cbx(self):
+        self.gmpes_npz = extract_npz(
+            self.session, self.hostname, self.calc_id, 'realizations',
+            message_bar=self.iface.messageBar(), params=None)
+        self.gmpes = self.gmpes_npz['array']['gsims']
+        self.gmpe_cbx.clear()
+        self.gmpe_cbx.setEnabled(True)
+        self.gmpe_cbx.addItems(self.gmpes)
+        self.rlzs_or_stats = [
+            self._format_rlz(gmpe_idx) for gmpe_idx in range(len(self.gmpes))]
```

Comment:
Do we need `_format_rlz` ? Just write ['rlz-%03d'  % r for r in range(len(rlzs_npz))`